code/modules/surgery/organs/augments_internal.dm

```diff
@@ -175,3 +175,125 @@
 	while(contents.len <= amount)
 		implant = pick(boxed)
 		new implant(src)
+
+//SkillChips
+
+/obj/item/organ/cyberimp/skillChip //Abstraction of the Skillchip Implants.
+	name = "SkillChip"
+	desc = "A piece of bleeding-edge tech which teaches the user various skills when installed."
+	slot = ORGAN_SLOT_SKILLCHIP
+	w_class = WEIGHT_CLASS_SMALL
+	var/applied_traits = list()
+	var/list/style = null
+
+/obj/item/organ/cyberimp/skillChip/Insert(mob/living/carbon/M, special, drop_if_replaced)
+	if(!iscarbon(M) || owner == M)
+		return
+
+	var/obj/item/organ/replaced = M.getorganslot(slot)
+	if(replaced)
+		replaced.Remove(M, special = 1)
+		if(drop_if_replaced)
+			replaced.forceMove(get_turf(M))
+		else
+			qdel(replaced)
+
+	SEND_SIGNAL(M, COMSIG_CARBON_GAIN_ORGAN, src)
+
+	owner = M
+	M.internal_organs |= src
+	M.internal_organs_slot[slot] = src
+	for(var/trait in applied_traits)
+		ADD_TRAIT(M, trait, "skillChip")
+	log_combat(M, M, "inserted this organ: ", "[name]")
+	moveToNullspace()
+	for(var/X in actions)
+		var/datum/action/A = X
+		A.Grant(M)
+	if(style)
+		for(var/martialArt in style)
+			var/datum/martial_art/S = new martialArt
+			S.teach(M)
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/organ/cyberimp/skillChip/Remove(mob/living/carbon/M, special)
+	owner = null
+	if(M)
+		M.internal_organs -= src
+		if(M.internal_organs_slot[slot] == src)
+			M.internal_organs_slot.Remove(slot)
+			for(var/trait in applied_traits)
+				REMOVE_TRAIT(M, trait, "skillChip")
+			if(style)
+				for(var/martialArt in style)
+					var/datum/martial_art/S = new martialArt
+					S.remove(M)
+			M.adjustOrganLoss(ORGAN_SLOT_BRAIN, 3)
+			to_chat(M, "<span class='warning>Your brain hurts as your neurons are forcefully rewired.</span>")
+		if((organ_flags & ORGAN_VITAL) && !special && !(M.status_flags & GODMODE))
+			M.death()
+	for(var/X in actions)
+		var/datum/action/A = X
+		A.Remove(M)
+
+	SEND_SIGNAL(M, COMSIG_CARBON_LOSE_ORGAN, src)
+
+	START_PROCESSING(SSobj, src)
```

Comment:
Do skillchips decay and expire when out of the body?

Reply:
Also could ..() be used here to clean up some repeated code?

Reply:
That comsig isn't organ decay. That comsig is actually to signal organ removal to M.

..() isn't viable for the remove process since I needed skillchips to have brain damage done only in line with removal being done and not when the outer most if gates fail but parent function completes then calls the brain damage and traits.